packages/cli/src/commands/install.js

```diff
@@ -1,83 +1,210 @@
+        return
+    }
+
+    try {
+        const paths = makePaths(cwd)
+        const providedOptions = computeProvidedOptions(groups)
+
+        const initialState = {
+            // will contain modules that should be imported and called
+            // A non empty array with falsy values will cause produce an import
+            // statement with no imported variables
+            support: [],
+
+            // will contain modules that should be imported and initialized
+            plugins: [],
+
+            // will be merged with existing contents of the cypress.json,
+            // result will overwrite existing contents.
+            // only if there's content
+            cypressJson: {},
+
+            // will be merged with existing contents of the cypress.env.json,
+            // result will overwrite existing contents.
+            // only if there's content
+            cypressEnvJson: {},
+
+            // will be merged with existing contents of the
+            // .cypress-cucumber-preprocessorrc.json, result will overwrite
+            // existing contents.
+            // only if there's content
+            cypressCucumberPreprocessorJson: {},
+        }
+
+        let state = initialState
+        state = await setBaseUrl({ options: providedOptions, state })
+        state = await setRecordVideo({ options: providedOptions, state })
+        state = await setProjectId({ options: providedOptions, state })
+        state = await setLoginUser({ options: providedOptions, state })
+        state = await setLoginPassword({ options: providedOptions, state })
+        state = await setLoginBackendUrl({ options: providedOptions, state })
+        state = await registerCommands({ options: providedOptions, state })
+        state = await setDataTestPrefix({
+            options: providedOptions,
+            state,
+            paths,
         })
-        .option('cucumber', {
-            describe: 'Configure Cucumber with Cypress',
-            type: 'boolean',
-            default: true,
+        state = await usePluginChromeAllowXSiteCookies({
+            options: providedOptions,
+            state,
         })
-        .option('support', {
-            describe: 'Setup DHIS2 support scripts for Cypress',
-            type: 'boolean',
-            default: true,
+        state = await usePluginCucumberPreprocessor({
+            options: providedOptions,
+            state,
         })
-        .option('force', {
-            describe: 'Overwrite existing configuration',
-            type: 'boolean',
-            default: false,
+        state = await setTestFilesToFeatureFiles({
+            options: providedOptions,
+            state,
         })
-        .option('verbose', {
-            describe: 'Detailed output',
-            type: 'boolean',
-            default: false,
+        state = await setNonGlobalStepDefinitions({
+            options: providedOptions,
+            state,
         })
```

Comment:
I think this is easier to read than it was before :+1: . If we want to isolate the mutation of `state`, we could wrap it in a function that builds out the state which would allow us to at least protect the `state` variable from unintentional reassignment in the `handler` scope.

E.g.

```js
const buildState = async (oldState, options) => {
    let newState = { ...oldState }
    newState = await setBaseUrl({ options, newState })
    newState = await setRecordVideo({ options, newState })
    newState = await setProjectId({ options, newState })
    newState = await setLoginUser({ options, newState })
    newState = await setLoginPassword({ options, newState })
    newState = await setLoginBackendUrl({ options, newState })
    newState = await registerCommands({ options, newState })
    newState = await setDataTestPrefix({
        options,
        newState,
        paths,
    })
    newState = await usePluginChromeAllowXSiteCookies({
        options,
        newState,
    })
    newState = await usePluginCucumberPreprocessor({
        options,
        newState,
    })
    newState = await setTestFilesToFeatureFiles({
        options,
        newState,
    })
    newState = await setNonGlobalStepDefinitions({
        options,
        newState,
    })

    return newState
}

const state = await buildState(initialState, providedOptions)
```

A minor change with a slight benefit, and if we do this it might be tempting to extract the `buildState` function into its own file. On one hand it is nice to be able to see the construction of the state in the handler without having to dig around for it, on the other hand, it may be more clear what the business logic is if it is extracted.

Reply:
I was thinking about this as well and, at least right now, prefer having it in the install command as is.
There's an alternative though: We could add the handler function to the options in `packages/cli/src/commands/install/options.js` and the just use a function (e. g. `processOptions`). The outcome would be more or less what you're suggesting, but it'd have the advantage that the option and its handler are coupled together / co-located in the code

Reply:
I'm OK with it as it is now, and we can refactor down the line if we need/want to.

Reply:
Too late